src/app/services/user.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable } from '@angular/core';
+import { Http, Headers, RequestOptions } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+import 'rxjs/add/operator/catch';
+import 'rxjs/add/operator/map';
+
+import { user } from '../model/user.interface';
+
+@Injectable()
+export class UserService {
+  private api = 'http://localhost:80/api/v1/user';
+
+  constructor(private http: Http) { }
+
+  save(user: user): any {
+    const headers = new Headers({ 'Content-Type': 'application/json' });
+    const options = new RequestOptions({ headers: headers });
+
+    return this.http.post(this.api, user, options)
+                .map(res => res.json())
+                .catch(this.handleError);
+
+  }
+
+  login(email: string, password: string): any {
+    const headers = new Headers({ 'Content-Type': 'application/json' });
+    const options = new RequestOptions({ headers: headers });
+    let body={
+        email:email,
+        password:password
+    }
+
+    return this.http.post(this.api+'/login', body, options)
+                .map(res => res.json())
+                .catch(this.handleError);
+
+  }
+  
+
+  private handleError (error: Response | any) {
```

Comment:
Error handling can be improved. `handleError` method will need to be copied where ever required. We can move this into central error handlers or interceptors.